source/adios2/helper/adiosMemory.inl

```diff
@@ -74,15 +84,55 @@ void InsertToBuffer(std::vector<char> &buffer, const T *source,
     buffer.insert(buffer.end(), src, src + elements * sizeof(T));
 }
 
+template <>
+inline void InsertToBuffer(std::vector<char> &buffer,
+                           const long double *source,
+                           const size_t elements) noexcept
+{
+    InsertToBufferSafe<long double>(buffer, source, elements);
+}
+
 template <class T>
-void CopyToBuffer(std::vector<char> &buffer, size_t &position, const T *source,
-                  const size_t elements) noexcept
+void CopyToBufferCommon(std::vector<char> &buffer, size_t &position,
+                        const T *source, const size_t elements) noexcept
 {
     const char *src = reinterpret_cast<const char *>(source);
     std::copy(src, src + elements * sizeof(T), buffer.begin() + position);
     position += elements * sizeof(T);
 }
 
+template <class T>
+void CopyToBufferSafe(std::vector<char> &buffer, size_t &position,
+                      const T *source, const size_t elements) noexcept
+{
+    // Create a temporary zero-initialized buffer which has the alignment we
+    // need
+    std::unique_ptr<T[]> tmp(new T[elements]);
+    std::memset(tmp.get(), 0, elements * sizeof(T));
```

Comment:
Can you drop the `memset` in lieu of the value-initialization constructor `new T[elements]()`?

Reply:
The problem is that value-initialization might not initialize the other 48 bits. We are attempting to explicitly fill those bits with 0.